bundles/org.openhab.core/src/main/java/org/openhab/core/internal/items/MetadataRegistryImpl.java

```diff
@@ -61,6 +64,17 @@ public boolean isInternalNamespace(String namespace) {
         return namespace.startsWith(INTERNAL_NAMESPACE_PREFIX);
     }
 
+    /**
+     * Provides all namespaces of a particular item
+     *
+     * @param itemname the name of the item for which the namespaces should be searched.
+     */
+    @Override
+    public Set<String> getAllNamespaces(String itemname) {
+        return getAll().stream().filter(n -> n.getUID().getItemName().equals(itemname))
+                .map(n -> n.getUID().getNamespace()).collect(Collectors.toSet());
+    }
```

Comment:
Did you see my comment regarding the mapping to the UID as first step?

Reply:
No sorry, I got overwhelmed with both you providing recommendations for the same line

Interesting that these two are supposedly the same :-) and too be honest I would have never come up with your way of doing it (took me quite a minute to understand even) but finally makes sense to me.


````
        Set<String> a=  stream().
                filter(n -> n.getUID().getItemName().equals(itemname))
                .map(n -> n.getUID().getNamespace()).collect(Collectors.toSet());
        
        Set<String> b=  stream().map(Metadata::getUID).
                filter(key -> key.getItemName().equals(itemname)).
                map(MetadataKey::getNamespace).collect(Collectors.toSet());
````